Occasional Background task crash in production

My Sentry instance is catching occasional failures to start the background task.

https://sentry.ecoeats.uk/share/issue/7cf6f7ee88ee454ca3e73647693f83e3/

The issue appears to be that the buildNotification method constructs an Intent but the constructor returns null. I'm unable to replicate this issue in a dev environment. 

`notificationIntent = new Intent(context, context.getCurrentActivity().getClass());`

Is the issue possibly that I'm attempting to start a task that is in the background already / the calling function is already in the background?

I'm seeing mostly the same bug: 

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4796)
  at android.app.ActivityThread.access$3100 (ActivityThread.java:308)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2346)
  at android.os.Handler.dispatchMessage (Handler.java:110)
  at android.os.Looper.loop (Looper.java:219)
  at android.app.ActivityThread.main (ActivityThread.java:8347)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:513)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1055)
Caused by: java.lang.NullPointerException: 
  at android.app.PendingIntent.getActivity (PendingIntent.java:350)
  at android.app.PendingIntent.getActivity (PendingIntent.java:317)
  at com.asterinet.react.bgactions.RNBackgroundActionsTask.buildNotification (RNBackgroundActionsTask.java:40)
  at com.asterinet.react.bgactions.RNBackgroundActionsTask.onStartCommand (RNBackgroundActionsTask.java:80)
  at android.app.ActivityThread.handleServiceArgs (ActivityThread.java:4776)
```

currently only reported on an Huawei Mate 20 with Android 10 (SDK 29)

--- a/android/src/main/java/com/asterinet/react/bgactions/RNBackgroundActionsTask.java
+++ b/android/src/main/java/com/asterinet/react/bgactions/RNBackgroundActionsTask.java
@@ -4,6 +4,7 @@ import android.app.Notification;
 import android.app.NotificationChannel;
 import android.app.NotificationManager;
 import android.app.PendingIntent;
+import android.content.Context;
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Build;
@@ -24,7 +25,7 @@ final public class RNBackgroundActionsTask extends HeadlessJsTaskService {
     private static final String CHANNEL_ID = "RN_BACKGROUND_ACTIONS_CHANNEL";
 
     @NonNull
-    public static Notification buildNotification(@NonNull final ReactContext context, @NonNull final BackgroundTaskOptions bgOptions) {
+    public static Notification buildNotification(@NonNull Context context, @NonNull final BackgroundTaskOptions bgOptions) {
         // Get info
         final String taskTitle = bgOptions.getTaskTitle();
         final String taskDesc = bgOptions.getTaskDesc();
@@ -35,7 +36,8 @@ final public class RNBackgroundActionsTask extends HeadlessJsTaskService {
         if (linkingURI != null) {
             notificationIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(linkingURI));
         } else {
-            notificationIntent = new Intent(context, context.getCurrentActivity().getClass());
+            //as RN works on single activity architecture - we don't need to find current activity on behalf of react context
+            notificationIntent = new Intent(Intent.ACTION_MAIN).addCategory(Intent.CATEGORY_LAUNCHER);
         }
         final PendingIntent contentIntent = PendingIntent.getActivity(context, 0, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
         final NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID)
@@ -54,7 +56,6 @@ final public class RNBackgroundActionsTask extends HeadlessJsTaskService {
             final boolean progressIndeterminate = progressBarBundle.getBoolean("indeterminate");
             builder.setProgress(progressMax, progressCurrent, progressIndeterminate);
         }
-
         return builder.build();
     }
 
@@ -77,7 +78,7 @@ final public class RNBackgroundActionsTask extends HeadlessJsTaskService {
         final BackgroundTaskOptions bgOptions = new BackgroundTaskOptions(extras);
         createNotificationChannel(bgOptions.getTaskTitle(), bgOptions.getTaskDesc()); // Necessary creating channel for API 26+
         // Create the notification
-        final Notification notification = buildNotification(getReactNativeHost().getReactInstanceManager().getCurrentReactContext(), bgOptions);
+        final Notification notification = buildNotification(this, bgOptions);
         startForeground(SERVICE_NOTIFICATION_ID, notification);
         return super.onStartCommand(intent, flags, startId);
     }

